nexus/src/app/background/phantom_disks.rs

```diff
@@ -0,0 +1,84 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Background task for detecting and un-deleting phantom disks
+//!
+//! A "phantom" disk is one where a disk delete saga partially completed but
+//! unwound: before a fix for customer-support#58, this would leave disks
+//! deleted but would also leave a `virtual_provisioning_resource` record for
+//! that disk. There would be no way to re-trigger the disk delete saga as the
+//! disk was deleted, so the project that disk was in could not be deleted.
+//!
+//! This background task detects that case and sets the disk to "faulted". The
+//! fix for custoemr-support#58 is similar: it changes the disk delete saga's
+//! unwind to also un-delete the disk and set it to faulted. This enables it to
```

Comment:
I could be misreading this, but this seems to imply that the changes to the disk-delete saga and this task have the same goal: Take a disk that could not be completely deleted; undelete it; set it to faulted; somehow notify the user that they can try again. Is that accurate? What I'm getting at is it seems that this task should not be needed if the saga is also doing this set of steps. It might be helpful to lay out why we need both of them.

Reply:
You're not wrong! The background task is there to catch previous cases where the saga unwound and caused these problems. Added this bit in 6e0827c13